build-logic/convention/build.gradle.kts

```diff
@@ -5,8 +5,9 @@ plugins {
 group = "com.skydoves.pokedex.compose.buildlogic"
 
 java {
-  sourceCompatibility = JavaVersion.VERSION_17
-  targetCompatibility = JavaVersion.VERSION_17
+  toolchain {
+    languageVersion.set(JavaLanguageVersion.of(17))
```

Comment:
This is a bad idea, it will force Gradle to use an old JDK to build the app, even if a newer one is available.
See https://jakewharton.com/gradle-toolchains-are-rarely-a-good-idea/